### ISSUE TITLE
CMS: Create VBA Editor role(s)

### ISSUE THREAD

**Issue by davidmpickett** [department-of-veterans-affairs/va.gov-cms#11334]
## Acceptance Criteria
- [ ] Testable_Outcome_X
- [ ] Testable_Outcome_Y
- [ ] Testable_Outcome_Z
- [ ] Requires design review

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [X] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by davidmpickett** [comment]
Surfaced as a need by @omahane https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11197#issuecomment-1289648669

**Comment by Dottisea** [comment]
Blocked by MVP definition

**Comment by davidmpickett** [comment]
These stories implicitly require having two workflows and therefore might require acting on this proposal: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12911

**Comment by omahane** [comment]
@davidmpickett It's not clear to me which content types will be getting the "Restricted archive workflow". Is it the following?

- VBA Facility
- VBA Facility Service
- Service Region

Additionally, the spec suggested adding "Service Region" to the workflow for a start? Do we want that in this ticket or the next?

**Comment by davidmpickett** [comment]
> @davidmpickett It's not clear to me which content types will be getting the "Restricted archive workflow". Is it the following?
> 
> * VBA Facility
> * VBA Facility Service
> * Service Region
> 

* VBA Facility - Yes - 100% for sure - this is the primary use case. But also, since those already have data, the migration process should probably be follow up ticket.
* VBA Facility Service - Maybe? 50/50 - I can see an argument either way. I wasn't intending those to be RAW, but they could be.
* Service Region - Kind of irrelevant. Since these types will only be managed by Content Admins or national editors - there's not a strong need for them to RAW, but it can't hurt. Also since it's a brand new content type, would be able to switch it over now with little to no risk and do some testing of workflow without waiting for future ticket when VBA Facilities switch over.


> Additionally, the spec suggested adding "Service Region" to the workflow for a start? Do we want that in this ticket or the next?

Up to you!

**Comment by omahane** [comment]
@xiongjaneg @mmiddaugh I assume we want to get rid of the COVID-related information here:
https://prod.cms.va.gov/help/va-benefit-facilities-vba/how-do-i-update-my-vba-facility-operating-status

I'm actually a little confused that we have another article very similar to the one above:
https://prod.cms.va.gov/how-to-edit-the-operating-status-of-a-va-benefits-facility-vba

**Comment by omahane** [comment]
@xiongjaneg @jilladams @davidmpickett Lastly, this ticket says to update documentation, but we do not have any documentation related to the new VBA content model or workflow. Is the expectation that we're creating the documentation for this?

**Comment by xiongjaneg** [comment]
@omahane The KB article https://prod.cms.va.gov/how-to-edit-the-operating-status-of-a-va-benefits-facility-vba is old and no longer linked to from the VBA KB landing page so I'll archive it.

I'll take a look at the COVID content in the other article and submit a revision as needed. Thank you!

**Comment by xiongjaneg** [comment]
@davidmpickett Do you recall why we added the AC: Documentation is updated and if so, where should that go?

**Comment by davidmpickett** [comment]
Pretty sure "Documentation is updated" is a standard AC for all tickets that introduce or change functionality. Places that have relevant documentation to review/update:

- https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/cms/content-governance
- https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/cms/user-administration
- https://prod.cms.va.gov/admin/structure/cm_document/note/112/VBA-Product-Notes
- https://prod.cms.va.gov/admin/structure/types/manage/vba_facility/document
- https://prod.cms.va.gov/admin/structure/types/manage/vba_facility_service/document

Places documentation may need to be created:
- Add a Content Model Document node.service_region - One of the new types for VBA, but currently no documentation
- Add a Content Model Document for module.core.workflows?
 - @swirtSJW not sure if this would be an appropriate place to add documentation about the two workflows until [this upstream issue](https://www.drupal.org/project/content_model_documentation/issues/3340432) is dealt with?

**Comment by omahane** [comment]
While looking at the phpunit tests, I found a permission that is set for both
`vamc_content_creator` and `content_creator_vet_center`.

Should we add this @swirtSJW ?
<img width="662" alt="Screenshot 2023-10-27 at 3 35 18 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/766573/61b9f6ed-0525-4465-8323-7318eabc9151">

**Comment by omahane** [comment]
Here's another one.
`assign content_creator_vba role` for `administrator_users`

<img width="902" alt="Screenshot 2023-10-27 at 3 41 56 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/766573/f147085c-3b14-443f-9d39-6a7e648ab2e5">

**Comment by davidmpickett** [comment]
Both good catches! Those both makes sense to me

**Comment by xiongjaneg** [comment]
This applies to VBA Facility content type (so they can't be archived and then brought back

**Comment by omahane** [comment]
@mmiddaugh Do you know what the process is/will be for creating VBA local and/or national editors? I'm looking at this documentation (for the purposes of updating, per this ticket):
https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/cms/user-administration

**Comment by xiongjaneg** [comment]
@omahane Michelle and I talked briefly about this. Very likely, we (I) will do a csv upload of the pilot MVP editors. We'll likely do the same as we did with Vet Center editors, which is start them out as content editors only and then after a national review and they've demonstrated they're comfortable, we can change their role to content publisher.

**Comment by omahane** [comment]
Thanks, @xiongjaneg for letting me know the results of your discussion. I don't think editors will be able to do what they really need to if they don't have the "Content Publisher" role assigned. The "Content Creator - VBA" role only gives a user the ability to create a VBA Facility Service. Here's an example screenshot of what it looks like for an editor without the "Content Publisher" role.
![Screenshot 2023-10-31 at 7 55 57 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/766573/787c0328-ef42-46d5-957e-473eaf93dc6a)
(cc: @davidmpickett to make sure I'm not off-base.)

**Comment by davidmpickett** [comment]
> Thanks, @xiongjaneg for letting me know the results of your discussion. I don't think editors will be able to do what they really need to if they don't have the "Content Publisher" role assigned. The "Content Creator - VBA" role only gives a user the ability to create a VBA Facility Service. Here's an example screenshot of what it looks like for an editor without the "Content Publisher" role. ![Screenshot 2023-10-31 at 7 55 57 AM](https://user-images.githubusercontent.com/766573/279382683-787c0328-ef42-46d5-957e-473eaf93dc6a.png) (cc: @davidmpickett to make sure I'm not off-base.)

Looks like you didn't actually assign this user the Editor Role
![Screenshot 2023-10-31 083029](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/02aaf3ef-720a-49f2-bd38-1e4116f9cf19)

**Comment by xiongjaneg** [comment]
Thanks for the clarification. Sounds like we can assign the content publisher role to start and "review" initial edits through work process rather than roles.

**Comment by omahane** [comment]
You're right, @davidmpickett . I just gave them content creator, not content editor. Thank you. I must have mis-read @xiongjaneg 's comment.

**Comment by jilladams** [comment]
Mid-sprint: Steve noted CMS should review PR.

**Comment by omahane** [comment]
> @omahane Michelle and I talked briefly about this. Very likely, we (I) will do a csv upload of the pilot MVP editors (method 2 from user admin, except to prod not training). We'll likely do the same as we did with Vet Center editors, which is start them out as content editors only and then after a national review and they've demonstrated they're comfortable, we can change their role to content publisher.

@xiongjaneg Just getting back to this, do you know what the process will be for account verification: https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/cms/user-administration#cms-account-verification

**Comment by xiongjaneg** [comment]
@omahane For now, the pilot editors' names are coming directly from VBA central and we're creating accounts for them.
